docs/Constraints.md

```diff
@@ -0,0 +1,123 @@
+#  Constraints
+
+| Constraint | `type` |
+|:-----|----------|
+| [Operating system](#operating-system) | `os` |
+| [Running template engine host](#template-engine-host) | `host` |
+
+The feature is available from .NET SDK 7.
+
```

Comment:
Do we want to link here the section from Reference-for-template.json.md that nicely describes the behavior of the feature?

Reply:
I added more info here.  Would like to avoid cross reference, as `Constraints` seems to be a master doc here.